test/functional/conftest.py

```diff
@@ -1,78 +1,68 @@
 # Note that this file is special in that py.test will automatically import this file and gather
 # its list of fixtures even if it is not directly imported into the corresponding test case.
-import pathlib
 
+import pathlib
 import pytest
+import sys
+import typing as ty
 
 import ipfshttpclient
 
 
 TEST_DIR = pathlib.Path(__file__).parent
 
 
-__is_available = None
-def is_available():  # noqa
+@pytest.fixture(scope='session')
+def ipfs_is_available() -> bool:
 	"""
 	Return whether the IPFS daemon is reachable or not
 	"""
-	global __is_available
 
-	if not isinstance(__is_available, bool):
-		try:
-			ipfshttpclient.connect()
-		except ipfshttpclient.exceptions.Error:
-			__is_available = False
-		else:
-			__is_available = True
+	try:
+		with ipfshttpclient.connect():
+			pass
+	except ipfshttpclient.exceptions.Error as e:
+		print('\nFailed to connect to IPFS client', file=sys.stderr)
+		print(e, file=sys.stderr)
 
-	return __is_available
+		return False
+	else:
+		return True
 
 
 def sort_by_key(items, key="Name"):
 	return sorted(items, key=lambda x: x[key])
 
 
-def get_client(offline=False):
-	if is_available():
-		return ipfshttpclient.Client(offline=offline)
+def _generate_client(
+		ipfs_is_available: bool,
+		offline: bool
+) -> ty.Generator[ipfshttpclient.Client, None, None]:
+	if ipfs_is_available:
+		with ipfshttpclient.Client(offline=offline) as client:
+			yield client
 	else:
 		pytest.skip("Running IPFS node required")
 
 
 @pytest.fixture(scope="function")
-def client():
-	"""Create a client with function lifetimme to connect to the IPFS daemon.
-
-	Each test function should instantiate a fresh client, so use this
-	fixture in test functions."""
-	with get_client() as client:
-		yield client
+def client(ipfs_is_available):
+	yield from _generate_client(ipfs_is_available, False)
 
 
 @pytest.fixture(scope="function")
 def offline_client():
-	"""Create a client in offline mode with function lifetimme"""
-	with get_client(offline=True) as client:
-		yield client
+	yield from _generate_client(ipfs_is_available, True)
 
 
 @pytest.fixture(scope="module")
 def module_client():
-	"""Create a client with a module lifetime to connect to the IPFS daemon.
-
-	For module-scope fixtures that need a client, if the client is to be created
-	automatically using a fixture (to keep client creation code centralized
-	here), that client-creating fixture must also be module-scope, so use
-	this fixture in module-scoped fixtures."""
-	with get_client() as client:
-		yield client
+	yield from _generate_client(ipfs_is_available, False)
 
 
 @pytest.fixture(scope="module")
 def module_offline_client():
```

Comment:
… aren't you missing a parameter named `ipfs_is_available` on this line to have pytest resolve the fixture?

Reply:
Yes, I am :sob: